File: php/types/strings/substring_modification.php
#!/usr/bin/php8.0
<?php

$cite = "Stat rosa pristina nomine, nomina nuda tenemus.";
print("cite = {$cite}\n\n");

$modified_cite = substr_replace($cite, "odonata", strpos($cite, "rosa"), strlen("rosa"));

print("modified cite = {$modified_cite}\n\n");

File: c++/types/strings/substring_modification.cpp
#include <iostream>
#include <cstring>

int main()
{
  std::string cite = "Stat rosa pristina nomine, nomina nuda tenemus.";
  std::cout << "cite = " << cite << std::endl << std::endl;

  std::string modified_cite = cite.replace(cite.find("rosa"), strlen("rosa"), "odonata");

  std::cout << "modified cite = " << modified_cite << std::endl << std::endl;
}
